src/Cimpress.Auth0.Client/Auth0TokenProvider.cs

```diff
@@ -1,88 +1,73 @@
 ﻿using System;
-using System.Collections.Concurrent;
 using System.Collections.Generic;
 using System.Net.Http;
 using System.Net.Http.Headers;
-using System.Text.RegularExpressions;
-using System.Threading;
 using System.Threading.Tasks;
 using Cimpress.Auth0.Client.Proxies;
 using Microsoft.Extensions.Logging;
 
 namespace Cimpress.Auth0.Client
 {
-    public class Auth0TokenProvider : IAuth0TokenProvider
+    public class Auth0TokenProvider : BaseAuth0TokenProvider, IAuth0TokenProvider
```

Comment:
I would like to see composition instead of inheritance.  Here I would suggest pulling out the code in the new `BaseAuth0TokenProvider` into something that can be injected into the `Auth0TokenProvider`, rather than having this provider and v2 provider both extend from the same class.  It will alievate some of the weirdness with the `autoscheduler` seen in in the v2 constructor.